tests/serv-minimal/src/servant_1.1.0/bench/servant_tb.cpp

```diff
@@ -0,0 +1,155 @@
+#include <stdint.h>
+#include <signal.h>
+
+#include "verilated_vcd_c.h"
+#include "Vservant_sim.h"
+
+using namespace std;
+
+static bool done;
+
+vluint64_t main_time = 0;       // Current simulation time
+// This is a 64-bit integer to reduce wrap over issues and
+// allow modulus.  You can also use a double, if you wish.
+
+double sc_time_stamp () {       // Called by $time in Verilog
+  return main_time;           // converts to double, to match
+  // what SystemC does
+}
+
+void INThandler(int signal)
+{
+	printf("\nCaught ctrl-c\n");
+	done = true;
+}
+
+typedef struct {
+  bool last_value;
+} gpio_context_t;
+
+void do_gpio(gpio_context_t *context, bool gpio) {
+  if (context->last_value != gpio) {
+    context->last_value = gpio;
+    printf("%lu output q is %s\n", main_time, gpio ? "ON" : "OFF");
+  }
+}
+
+typedef struct {
+  uint8_t state;
+  char ch;
+  uint32_t baud_t;
+  vluint64_t last_update;
+} uart_context_t;
+
+void uart_init(uart_context_t *context, uint32_t baud_rate) {
+  context->baud_t = 1000*1000*1000/baud_rate;
+  context->state = 0;
+}
+
+bool do_uart(uart_context_t *context, bool rx) {
+  if (context->state == 0) {
+    if (rx)
+      context->state++;
+  }
+  else if (context->state == 1) {
+    if (!rx) {
+      context->last_update = main_time + context->baud_t/2;
+      context->state++;
+    }
+  }
+  else if(context->state == 2) {
+    if (main_time > context->last_update) {
+      context->last_update += context->baud_t;
+      context->ch = 0;
+      context->state++;
+    }
+  }
+  else if (context->state < 11) {
+    if (main_time > context->last_update) {
+      context->last_update += context->baud_t;
+      context->ch |= rx << (context->state-3);
+      context->state++;
+    }
+  }
+  else {
+    if (main_time > context->last_update) {
+      context->last_update += context->baud_t;
+      context->state=1;
+      return true;
+    }
+  }
+  return false;
+}
+
+int main(int argc, char **argv, char **env)
+{
+  vluint64_t sample_time = 0;
+	uint32_t insn = 0;
+	uint32_t ex_pc = 0;
```

Comment:
The indentation is too long

Reply:
Fixed - replaced mixed tabs and spaces with spaces only.